staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -21,39 +21,254 @@ import (
 
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
 )
 
-// profileLegacy represents the legacy debugging profile which is backwards-compatible with 1.23 behavior.
-func profileLegacy(pod *corev1.Pod, containerName string, target runtime.Object) error {
+type legacyProfile struct {
+}
+
+type generalProfile struct {
+}
+
+type baselineProfile struct {
+}
+
+type restrictedProfile struct {
+}
+
+func (p *legacyProfile) Apply(pod *corev1.Pod, containerName string, target runtime.Object) error {
 	switch target.(type) {
 	case *corev1.Pod:
 		// do nothing to the copied pod
 		return nil
 	case *corev1.Node:
-		const volumeName = "host-root"
-		pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
-			Name: volumeName,
-			VolumeSource: corev1.VolumeSource{
-				HostPath: &corev1.HostPathVolumeSource{Path: "/"},
-			},
+		MountRootPartition(pod, containerName)
+		UseHostNamespaces(pod)
+		return nil
+	default:
+		return fmt.Errorf("the %s profile doesn't support objects of type %T", ProfileLegacy, target)
+	}
+}
+
+type debugStyle int
+
+const (
+	// debug by ephemeral container
+	styleEphemeral debugStyle = iota
+	// debug by pod copy
+	stylePodCopy
+	// debug node
+	styleNode
+)
+
+func getDebugStyle(pod *corev1.Pod, target runtime.Object) (debugStyle, error) {
+	switch target.(type) {
+	case *corev1.Pod:
+		if asserted, ok := target.(*corev1.Pod); ok {
+			if pod != asserted { // comparing addresses
+				return stylePodCopy, nil
+			}
+		}
+		return styleEphemeral, nil
+	case *corev1.Node:
+		return styleNode, nil
+	}
+	return debugStyle(-1), fmt.Errorf("objects of type %T are not supported", target)
```

Comment:
I know I said this multiple times, but this `debugStyle(-1)` is not gopher way :)

Reply:
Oops - replaced it with `styleUnsupported` in [2111c5c](https://github.com/kubernetes/kubernetes/pull/114280/commits/2111c5cc670aff0d4a0a19285833bfb603f48188)